ckcp/openshift_dev_setup.sh

```diff
@@ -146,6 +154,15 @@ init() {
   fi
   cp -rf "$GITOPS_DIR/sre" "$WORK_DIR"
   update_git_reference "$GIT_URL" "$GIT_REF" "$WORK_DIR/environment/kcp/kustomization.yaml"
+  {
+    echo "---"
+    kustomize build ckcp/work/environment/kcp/workspace-controller/base | \
```

Comment:
This should not be the way to patch kustomize files. We should try to use overlay/kustomization.yaml to modify values as that is the purpose of overlay folder.
We could have two folders under overlay called dev and staging.
overlays/
  - dev
    - kustomization.yaml
  - staging
    - kustomization.yaml
dev could have the value provided by user.
staging could have the value from our staging cluster.

Reply:
@bnallapeta ops(stable/staging) and dev should not share the same WORK_DIR in the first place I think so we don't need dev and staging folders here.
However I agree that using `sed -e` anywhere in this project seems like an anti-pattern to kustomize and we may rather want to patch via kustomize rather than sed.

Reply:
After further looking into kustomize it seems it [does not cope well with vars](https://github.com/kubernetes-sigs/kustomize/issues/4120)

Reply:
I rewrote that bit and moved it to `setup_kcp.sh` as the patching is required for SRE as well. Let's track follow-up feedback on a new comment if necessary.

Reply:
Since the change is only related to ckcp, it makes sense to include the patch in setup_kcp.sh. Thanks.